crates/binjs_io/src/context/prelude.rs

```diff
@@ -0,0 +1,214 @@
+use context::Name;
+use TokenReaderError;
+
+use bytes::varnum::ReadVarNum;
+use util::PosRead;
+
+use std::io::Read;
+
+/// A compression format used in the prelude.
+#[derive(Clone, Debug)]
+pub enum Compression {
+    /// The section is compressed using brotli.
+    Brotli,
+    /// The section is uncompressed.
+    Identity,
+}
+impl Compression {
+    /// Return the representation as a 4 bytes name.
+    pub fn as_bytes(&self) -> &'static Name {
+        match *self {
+            Compression::Brotli => b"BROT",
+            Compression::Identity => b"NONE",
+        }
+    }
+
+    /// Parse the 4 bytes name.
+    pub fn from_bytes(bytes: &Name) -> Option<Compression> {
+        match bytes {
+            b"BROT" => Some(Compression::Brotli),
+            b"NONE" => Some(Compression::Identity),
+            _ => None,
+        }
+    }
+}
+
+/// The name of a section of the prelude.
+#[derive(Clone, Debug)]
+pub enum Section {
```

Comment:
I was thinking that prelude/content/main are called Section, and each dictionary is something like SubSection.
is it wrong?

Reply:
Oops, SubSection, you're right.